File: android/src/main/kotlin/tech/shmy/a_player/player/APlayerUtil.kt
package tech.shmy.a_player.player

class APlayerUtil {
    companion object {
        fun isHttpProtocol(url: String): Boolean {
            return url.startsWith("http://") || url.startsWith("https://")
        }
        fun isFileProtocol(url: String): Boolean {
            return url.startsWith("file://")
        }
        fun isAssetProtocol(url: String): Boolean {
            return !isHttpProtocol(url) && !isFileProtocol(url)
        }
        fun isUserAgentKey(key: String): Boolean {
            return key.uppercase() == "USER-AGENT"
        }
        fun isRefererKey(key: String): Boolean {
            return key.uppercase() == "REFERER"
        }
    }
}

File: ios/Classes/Player/APlayerUtil.swift
//
//  APlayerUtil.swift
//  a_player
//
//  Created by drm-chao wang on 2022/7/5.
//

import Foundation

class APlayerUtil {
    static func isHttpProtocol(url: String) -> Bool {
        return url.starts(with: "http://") || url.starts(with: "https://")
    }
    static func isFileProtocol(url: String) -> Bool {
        return url.starts(with: "file://")
    }
    static func isUserAgentKey(key: String) -> Bool {
        return key.uppercased() == "USER-AGENT"
    }
    static func isRefererKey(key: String) -> Bool {
        return key.uppercased() == "REFERER"
    }
}
